include/utils.hpp

```diff
@@ -104,6 +104,44 @@ operator+(const fixed_string<CharT, N1> &lhs, const fixed_string<CharT, N2> &rhs
     return result;
 }
 
+namespace detail {
+constexpr int
+log10(int n) noexcept {
+    if (n < 10) return 0;
+    return 1 + log10(n / 10);
+}
+
+constexpr int
+pow10(int n) noexcept {
+    if (n == 0) return 1;
+    return 10 * pow10(n - 1);
```

Comment:
Should these math helper functions fall back to a non recursive implementation if they are evaluated at runtime? Or be `consteval` or an assertion to prevent accidental runtime use?